From 840c15e99abc0e2ca6ca8ec6ea871814f227ca06 Mon Sep 17 00:00:00 2001
From: Meron Cohen <mecohen@habana.ai>
Date: Fri, 04 Apr 2025 00:52:45 +0300
Subject: [PATCH] [SW-224519] common: fix utilization function

When calling hl-smi with -Q for querying utiliztion.aip, the displayed
value could be corrupted. However, when calling hl-smi without -Q, then
the utilization was displayed correctly. The reason is that in the former
case, the power drawn from the secondary PSU is not getting updated, while
in the latter case it is. Therefore, there was a need to update it in the
utilization function: hlml_device_get_utilization_rates.

Change-Id: I16d49dfba7a530431e43102d091de0233e386dcf
Signed-off-by: Meron Cohen <mecohen@habana.ai>
---

diff --git a/src/hlml.c b/src/hlml.c
index d40d274..3f002ea 100644
--- a/src/hlml.c
+++ b/src/hlml.c
@@ -460,6 +460,12 @@
 	}
 	aip_info->power = (unsigned int)MW_TO_WATT(power_consumption);
 
+	ret_int = hl_smi_sensors_get_info(aip_info, HL_SMI_SENSORS_INFO_SEC_POW);
+	if (ret_int != HLML_INT_SUCCESS) {
+		ret = HLML_ERROR_NO_DATA;
+		goto dev_get_util_err;
+	}
+
 	if (aip_info->use_new_utilization_formula) {
 		/*
 	 	* The utilization formula is:
